test/byContract/IexecPoco/04_finalize.test.ts

```diff
@@ -0,0 +1,331 @@
+            const { callbackResultDigest: wrongCallbackResultDigest } =
+                buildResultCallbackAndDigest(567);
+            const workers = [
+                { wallet: worker1, callbackResultDigest: callbackResultDigest },
+                { wallet: worker2, callbackResultDigest: wrongCallbackResultDigest },
+                { wallet: worker3, callbackResultDigest: callbackResultDigest },
+                { wallet: worker4, callbackResultDigest: callbackResultDigest },
+            ];
+            const winningWorkers = [worker1, worker3, worker4];
+            const losingWorker = worker2;
+            for (const worker of workers) {
+                const { resultHash, resultSeal } = buildResultHashAndResultSeal(
+                    taskId,
+                    worker.callbackResultDigest,
+                    worker.wallet,
+                );
+                const schedulerSignature = await buildAndSignContributionAuthorizationMessage(
+                    worker.wallet.address,
+                    taskId,
+                    enclaveAddress,
+                    scheduler,
+                );
+                await iexecWrapper.depositInIexecAccount(worker.wallet, workerTaskStake);
+                await iexecPoco
+                    .connect(worker.wallet)
+                    .contribute(
+                        taskId,
+                        resultHash,
+                        resultSeal,
+                        enclaveAddress,
+                        '0x',
+                        schedulerSignature,
+                    )
+                    .then((tx) => tx.wait());
+            }
+            for (const worker of winningWorkers) {
+                await iexecPoco
+                    .connect(worker)
+                    .reveal(taskId, callbackResultDigest)
+                    .then((tx) => tx.wait());
+            }
+            expect(await iexecPoco.balanceOf(requester.address)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(requester.address)).to.be.equal(0);
+            expect(await iexecPoco.balanceOf(sponsor.address)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(sponsor.address)).to.be.equal(dealPrice);
+            expect(await iexecPoco.balanceOf(appProvider.address)).to.be.equal(0);
+            expect(await iexecPoco.balanceOf(datasetProvider.address)).to.be.equal(0);
+            expect(await iexecPoco.balanceOf(scheduler.address)).to.be.equal(0);
+            expect(await iexecPoco.frozenOf(scheduler.address)).to.be.equal(schedulerDealStake);
+            for (const worker of [...winningWorkers, losingWorker]) {
+                expect(await iexecPoco.balanceOf(worker.address)).to.be.equal(0);
+                expect(await iexecPoco.frozenOf(worker.address)).to.be.equal(workerTaskStake);
+                expect(await iexecPoco.viewScore(worker.address)).to.be.equal(0);
+            }
+            expect(await iexecPoco.balanceOf(iexecPoco.address)).to.be.equal(
```

Comment:
Put the test of iexecPoco.address at the top of other tests ?

Reply:
Moved both ok!